Src/FluentAssertions/Specialized/ActionAssertions.cs

```diff
@@ -114,6 +114,54 @@ public void NotThrow(string because = "", params object[] becauseArgs)
             }
         }
 
+        /// <summary>
+        /// Asserts that the current <see cref="Action"/> stops throwing any exception
+        /// after a specified amount of time.
+        /// </summary>
+        /// <remarks>
+        /// The <see cref="Action"/> is invoked. If it raises an exception,
+        /// the invocation is repeated until it either stops raising any exceptions
+        /// or the specified wait time is exceeded.
+        /// </remarks>
+        /// <param name="waitTime">
+        /// The time in milliseconds after which the <see cref="Action"/> should have stopped throwing any exception.
+        /// </param>
+        /// <param name="pollInterval">
+        /// The time between subsequent invocations of the <see cref="Action"/>.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        /// <exception cref="ArgumentOutOfRangeException">Throws if waitTime or pollInterval are negative.</exception>
+        public void NotThrowAfter(int waitTime, int pollInterval, string because = "", params object[] becauseArgs)
```

Comment:
🔧 I would prefer to use `TimeSpan` for `waitTime` and `pollInterval`.

Reply:
🔧 Maybe `NotThrowAnymoreAfter`

Reply:
🤔 Do we also need to be able to specify an exception type here?
🤔 Do we also need this assertion for `Func<Task>` for consistency with the other exception APIs?

Reply:
> Maybe `NotThrowAnymoreAfter`

Are you sure? I think that this would be a tad too long. It might also be misleading. Doesn't "anymore" suggest that the Action must have been throwing exceptions at some point

Reply:
> I would prefer to use `TimeSpan` for `waitTime` and `pollInterval`.
Sounds good. Would you keep an overload with the current signature or would you expect the user to convert from that representation to the TimeSpan?

Reply:
True. Very true. Alright. Let's stick with `NotThrowAfter`

Reply:
> Would you keep an overload with the current signature or would you expect the user to convert from that representation to the TimeSpan?

They can use the `TimeSpan` extension methods like `12.Seconds()`.

Reply:
> Do we also need to be able to specify an exception type here?

I have not encountered a use case for this so far. This would mean that the Action could be still throwing exceptions but not the specified one ... :confused. We could add it for consistency's sake. You could also wait until someone requests it. I don't know your policy regarding these kind of questions :smirk:
> Do we also need this assertion for `Func<Task>` for consistency with the other exception APIs?
Probably.